modules/nlu/src/backend/storage.ts

```diff
@@ -19,20 +19,22 @@ export interface AvailableModel {
 }
 
 export default class Storage {
-  private ghost: ScopedGhostService
-  private intentsDir: string
-  private entitiesDir: string
-  private modelsDir: string
+  static ghostProvider: (botId: string) => sdk.ScopedGhostService
+
+  private readonly ghost: ScopedGhostService
+  private readonly intentsDir: string
+  private readonly entitiesDir: string
+  private readonly modelsDir: string
 
-  constructor(bp: SDK, config, botId) {
-    this.ghost = bp.ghost.forBot(botId)
+  constructor(config: Config, private readonly botId: string) {
```

Comment:
ins't everything public by default ? why bloat the code with the public keyword all around ?

Reply:
+1 about not adding public everywhere. Should be clear enough if we use private and start with underscore

Reply:
I think its my dark C# past surfacing. #helpme